source/common/router/router.cc

```diff
@@ -935,6 +1060,8 @@ bool Filter::setupRedirect(const Http::HeaderMap& headers) {
 void Filter::doRetry() {
   is_retry_ = true;
   attempt_count_++;
+  ASSERT(pending_retries_ > 0);
```

Comment:
just make `pending_retries_` unsigned to avoid it going below 0

Reply:
my intention here was to blow up instead of overflow, but i guess i could `assert > 0` before decrementing

Reply:
oh i do do that lol